### ISSUE TITLE
Money Manager Ex Crashed After Deleting Transaction

### ISSUE THREAD

**Issue by Ocirasai** [moneymanagerex/moneymanagerex#6412]
MMEX version:
 - [ ] 1.6.4
 - [x] 1.7.0
 - [ ] 1.7.1
 - [ ] 1.7.2
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

In an account I use to track one of my utility bills, I had copied 4 previous transactions, changed the date of the copied transactions (one at a time) to the date on which they occurred, and realized one of the transactions wasn't needed, so I chose Delete > Delete selected transaction, and then program crashed. The only thing I noted was that the transaction I tried to delete was marked as Reconciled (R), but this does not seem to make a difference.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Copy a transaction
2. Change the date on the transaction
3. Attempt to delete the transaction

Expected result:

The selected transaction is deleted without issue.

Actual result:

The program crashes.

# Additional information

If you have a backtrace for a crash or a warning, paste it here.
[MoneyManagerEx_dbgrpt-14852-20240101T015012.zip](https://github.com/moneymanagerex/moneymanagerex/files/13803792/MoneyManagerEx_dbgrpt-14852-20240101T015012.zip)

**Comment by whalley** [comment]
When you reopened the DB was the deleted transaction still there, if not, was it in the 'Deleted Transactions' list?
Is it just this transaction that causes the crash on deletion? What if you delete other transactions?

Does the deletion work if you use the delete button at the bottom of the screen?

https://github.com/moneymanagerex/moneymanagerex/issues/6139

**Comment by Ocirasai** [comment]
When I first re-opened the program, an MMEX Instance Check Dialog Box appeared asking me to confirm that I wanted to open the database.

When I went back into the account, I did not see the transaction listed, so I believe the transaction did get deleted.

I'm not sure where the Deleted Transactions list is, so if you can let me know where to look, I can reproduce the issue and look to see if it is still there.

If I select Delete at the bottom instead of using the menu option, the program does not close/crash, but I'm brought back to the Dashboard, which means I then need to select the account again to go back in and continue adding transactions.

**Comment by whalley** [comment]
OK, sounds like duplication of https://github.com/moneymanagerex/moneymanagerex/issues/6139, so I'll close this as a duplicate. 

@Ocirasai  If you can supply a cutdown copy of a database that exhibits this problem then that would be useful to help debug the issue.

**Comment by Ocirasai** [comment]
I created a Test Database and only added one account with one transaction in it.

My steps to reproduce the issue were to:

1. Copy the existing transaction (Using Ctrl-C and then Ctrl-V).
2. Right click on the new transaction and then select Delete > Delete selected transaction...

I've attached the sample database I used along with the crash dump that was created.

Thank you.

[Test.zip](https://github.com/moneymanagerex/moneymanagerex/files/13806255/Test.zip)
